src/System.Net.NameResolution/src/System/Net/DNS.cs

```diff
@@ -554,6 +571,9 @@ public static IPAddress[] GetHostAddresses(string hostNameOrAddress)
         public static IAsyncResult BeginGetHostEntry(string hostNameOrAddress, AsyncCallback requestCallback, object stateObject)
         {
             if (NetEventSource.Log.IsEnabled()) NetEventSource.Enter(NetEventSource.ComponentType.Socket, "DNS", "BeginGetHostEntry", hostNameOrAddress);
+
+            NameResolutionPal.EnsureSocketsAreInitialized();
```

Comment:
For maintainability, I like keeping these at the top level.  Someone could add a NameResolutionPal call before the helper call later, and forget to add the EnsureSocketsAreInitialized call...